felix/ipsets/ipsets.go

```diff
@@ -801,6 +833,9 @@ func (s *IPSets) tryUpdates(ipsetFilter func(ipSetName string) bool, programmedI
 			"stderr":     s.stderrCopy.String(),
 			"input":      s.restoreInCopy.String(),
 		}).Warning("Failed to complete ipset restore, IP sets may be out-of-sync.")
+		for _, setName := range touchedIPSets {
```

Comment:
Doesn't it mean, that you then retry all IP sets up to the failure ...  but what about those that were not touched yet? Don't all of them remain dirty so they all would be retried anyway?

Reply:
Intention is to do a resync of the ones that we "touched" before we finish updating all the dirty IP sets.  We don't need to resync all dirty IP sets because the ones that we haven't touched yet should still be in the expected state.  The problem with the IP sets protocol is that we can't tell which changes were successful in the batch that we just tried so we have to resync all the IP sets that we might have modified.